modules/php/Actions/Scenario2/ProgramRobot.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Bga\Games\WelcomeToTheMoon\Actions\Scenario2;
+
+use Bga\Games\WelcomeToTheMoon\Core\Globals;
+use Bga\Games\WelcomeToTheMoon\Core\Notifications;
+use Bga\Games\WelcomeToTheMoon\Models\Player;
+
+class ProgramRobot extends \Bga\Games\WelcomeToTheMoon\Models\Action
+{
+  public function getState(): int
+  {
+    return ST_PROGRAM_ROBOT;
+  }
+
+  public function argsProgramRobot()
+  {
+    $player = $this->getPlayer();
+
+    return [
+      'slots' => $player->scoresheet()->getSectionFreeSlots('robots'),
+    ];
+  }
+
+  /**
+   * @throws \BgaVisibleSystemException
+   */
+  public function actProgramRobot(int $slot)
+  {
+    $player = $this->getPlayer();
+    $scoresheet = $player->scoresheet();
+
+    $scribble = $scoresheet->addScribble($slot);
+    Notifications::programRobot($player, $scribble);
+
+    $multiplier = self::getMultiplierToAchieve($player, $slot);
+    if (!is_null($multiplier)) {
+      if ($scoresheet->hasScribbledSlot($multiplier)) {
+        // Circling small multiplier instead
+        $multiplier = self::$bigToSmallMultiplierMap[$multiplier];
+      } else {
+        // Register for phase 5
+        $multipliers = Globals::getCircledMultipliers();
+        $multipliers[] = $multiplier;
+        Globals::setCircledMultipliers($multipliers);
+      }
+      $scribble = $scoresheet->addScribble($multiplier, SCRIBBLE_CIRCLE);
+      Notifications::circleMultiplier($player, $scribble, self::getMultiplierValue($multiplier));
+    }
+  }
+
+  /**
+   * Find the big multiplier assigned to this robot slot id
+   * @throws \BgaVisibleSystemException
+   */
+  private function getMultiplierToAchieve(Player $player, int $robotSlotId): int | null
+  {
+    $scoresheet = $player->scoresheet();
+    $multiplierAffected = self::findMultiplierByRobot($robotSlotId);
+
+    $scribbledMap = array_map(function ($slotId) use ($scoresheet) {
+      return $scoresheet->hasScribbledSlot($slotId);
+    }, self::$bigMultipliers[$multiplierAffected]);
+
+    return in_array(false, $scribbledMap) ? null : $multiplierAffected;
+  }
+
+  private function findMultiplierByRobot(int $robotSlotId): int
+  {
+    foreach (self::$bigMultipliers as $multiplier => $robots) {
+      if (in_array($robotSlotId, $robots)) {
+        return $multiplier;
+      }
+    }
+    throw new \BgaVisibleSystemException('findMultiplierByRobot: no multipliers found for slot id ' . $robotSlotId);
+  }
+
+  public static function getMultiplierValue(int $multiplierId): int
```

Comment:
This method is used here and in Scoresheet2.php. Do you think it's fine to store this array here or do we have some common place for this?